hazelcast/src/main/java/com/hazelcast/jet/impl/ClientJobProxy.java

```diff
@@ -135,7 +135,7 @@ protected UUID findLightJobCoordinator() {
 
     @Override
     protected CompletableFuture<Void> invokeJoinJob() {
-        ClientMessage request = JetJoinSubmittedJobCodec.encodeRequest(getId(), isLightJob());
+        ClientMessage request = JetJoinSubmittedJobCodec.encodeRequest(getId(), lightJobCoordinator);
         ClientInvocation invocation = invocation(request, masterId());
```

Comment:
Why are we still sending all of these invocations to the master?

Reply:
I'm just wondering what would be a problem if we hadn't changed any codecs and direct these operations to the light job coordinator here.

Reply:
That was the issue before - it works for smart clients. In a non-smart clients if you ask the message to be sent to some member, it will still be sent to the member the client is connected to.
I expect even for smart clients when there's a connection glitch, it can be delivered to some other member.

Reply:
Thanks for the explanation. For decreasing the number of hops on the happy path, wdyt about trying to forward these client invocations to the corresponding coordinators first? (You can ignore this negligible minor optimization)

Reply:
Good point, [fixed that](https://github.com/hazelcast/hazelcast/pull/19010/commits/87c74e624f0f0e09de28c9fee6c32b4cc32aed98), it was an omission.